src/firebase/firestore/index.ts

```diff
@@ -42,43 +40,17 @@ export const getUserRoleData = async (uid: string | undefined): Promise<string |
 	}
 }
 
-/**
- * Function creates a new document in "users" collection where sets all new user's data;
- *
- * @param {ROLES} role - new user's role in the system (admin or garson);
- * @param {string} uid - new user's uid;
- */
-export const createNewUser = async (userData: NewUserProps): Promise<void> => {
+export const addUserToCollection = async (userData: INewUserProps): Promise<void> => {
```

Comment:
with this simplicity I'm not sure that this function needs to exist... do we really create a user like that anywhere in the project?

consider trying to remove the function and just place the code you have in `try` block to the funtion that creates user with email and password.